cstar/base/input_dataset.py

```diff
@@ -58,7 +58,7 @@ def __init__(
         self.base_model: "BaseModel" = base_model
         self.source: DataSource = DataSource(location)
         self.file_hash: Optional[str] = file_hash
-        self.working_path: Optional[Path] = None
+        self.working_path: Optional[Path | List[Path]] = None
```

Comment:
Since python 3.9 you have been able to use the built-in collections for typing, so you can use `list` instead of `typing.List`
```suggestion
        self.working_path: Optional[Path | list[Path]] = None
```